router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming_test.go

```diff
@@ -310,6 +337,101 @@ func TestSnowpipeStreaming(t *testing.T) {
 			"status":        "failed",
 		}).LastValue())
 	})
+
+	t.Run("Upload with unauthorized schema error should add backoff", func(t *testing.T) {
+		statsStore, err := memstats.New()
+		require.NoError(t, err)
+
+		sm := New(config.New(), logger.NOP, statsStore, destination)
+		sm.channelCache.Store("RUDDER_DISCARDS", rudderDiscardsChannelResponse)
+		sm.api = &mockAPI{
+			createChannelOutputMap: map[string]func() (*model.ChannelResponse, error){
+				"USERS": func() (*model.ChannelResponse, error) {
+					return &model.ChannelResponse{Code: internalapi.ErrSchemaDoesNotExistOrNotAuthorized}, nil
+				},
+			},
+		}
+		managerCreatorCallCount := 0
+		sm.managerCreator = func(_ context.Context, _ whutils.ModelWarehouse, _ *config.Config, _ logger.Logger, _ stats.Stats) (manager.Manager, error) {
+			sm := snowflake.New(config.New(), logger.NOP, stats.NOP)
+			managerCreatorCallCount++
+			mockManager := newMockManager(sm)
+			mockManager.createSchemaErr = fmt.Errorf("failed to create schema")
+			return mockManager, nil
```

Comment:
Can we use different names here as these are getting shadowed? Also, it's confusing we are using sm for for `snowpipestreaming.Manager` and `snowflake.Snowflake`
```suggestion
			managerCreatorCallCount++
			
			s := snowflake.New(config.New(), logger.NOP, stats.NOP)

			mm := newMockManager(s)
			mm.createSchemaErr = fmt.Errorf("failed to create schema")
			return mm, nil
```

Reply:
Same for the below test as well.

Reply:
Renamed in both the tests